sklearn/cluster/_dbscan_inner.pyx

```diff
@@ -6,36 +6,72 @@ cimport cython
 from libcpp.vector cimport vector
 cimport numpy as np
 import numpy as np
+from scipy import sparse
 
 
 # Work around Cython bug: C++ exceptions are not caught unless thrown within
 # a cdef function with an "except +" declaration.
 cdef inline void push(vector[np.npy_intp] &stack, np.npy_intp i) except +:
     stack.push_back(i)
 
+def query_nn(pt, neigh, eps, sample_weight):
+    pt_neighborhoods = neigh.radius_neighbors(pt, eps, return_distance=False)
+    if sample_weight is None:
+        n_neighbors = len(pt_neighborhoods[0])
+    else:
+        n_neighbors = np.sum(sample_weight[pt_neighborhoods[0]])
+
+    return (pt_neighborhoods[0], n_neighbors)
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
 def dbscan_inner(np.ndarray[np.uint8_t, ndim=1, mode='c'] is_core,
                  np.ndarray[object, ndim=1] neighborhoods,
-                 np.ndarray[np.npy_intp, ndim=1, mode='c'] labels):
+                 np.ndarray[np.npy_intp, ndim=1, mode='c'] labels,
+                 X,
+                 eps,
+                 min_samples,
+                 sample_weight,
+                 neigh,
```

Comment:
I think we should just pass in `get_neighborhood()` so this isn't littered with the parameters for `query_nn`